DatadogFlags/Sources/FlagsClient.swift

```diff
@@ -20,20 +20,126 @@ public class FlagsClient {
+                try createOrThrow(with: configuration, instanceName: name, in: core)
             }
+
+            // Register the created client in the registry
+            FlagsClientRegistry.register(client, named: name)
+            return client
         } catch let error {
-            consolePrint("\(error)", .error)
-            fatalError("TODO: FFL-1016 Fallback to NOP Client")
+            DD.logger.error("Failed to create FlagsClient with name '\(name)'", error: error)
+            return NOPFlagsClient()
         }
     }
 
-    internal static func createOrThrow(with configuration: FlagsClient.Configuration, in core: DatadogCoreProtocol) throws -> FlagsClient {
+    /// Creates a FlagsClient instance with default configuration, or returns existing if one with the same name exists.
+    ///
+    /// - Parameter core: The DatadogCore instance to use. Defaults to CoreRegistry.default.
+    /// - Returns: FlagsClient instance (new or existing).
+    public static func createOrGet(in core: DatadogCoreProtocol = CoreRegistry.default) -> FlagsClient {
+        return createOrGet(with: Configuration(), name: FlagsClientRegistry.defaultInstanceName, in: core)
+    }
+
+    /// Creates a named FlagsClient instance with default configuration, or returns existing if one with the same name exists.
+    ///
+    /// - Parameters:
+    ///   - name: The unique name for this instance.
+    ///   - core: The DatadogCore instance to use. Defaults to CoreRegistry.default.
+    /// - Returns: FlagsClient instance (new or existing).
+    public static func createOrGet(name: String, in core: DatadogCoreProtocol = CoreRegistry.default) -> FlagsClient {
+        return createOrGet(with: Configuration(), name: name, in: core)
+    }
+
+    /// Creates a FlagsClient instance with custom configuration, or returns existing if one with the same name exists.
+    ///
+    /// - Parameters:
+    ///   - configuration: Custom configuration for the client.
+    ///   - core: The DatadogCore instance to use. Defaults to CoreRegistry.default.
+    /// - Returns: FlagsClient instance (new or existing).
+    public static func createOrGet(with configuration: FlagsClient.Configuration, in core: DatadogCoreProtocol = CoreRegistry.default) -> FlagsClient {
+        return createOrGet(with: configuration, name: FlagsClientRegistry.defaultInstanceName, in: core)
+    }
+
+    /// Creates a named FlagsClient instance with custom configuration, or returns existing if one with the same name exists.
+    ///
+    /// - Parameters:
+    ///   - configuration: Custom configuration for the client.
+    ///   - name: The unique name for this instance.
+    ///   - core: The DatadogCore instance to use. Defaults to CoreRegistry.default.
+    /// - Returns: FlagsClient instance (new or existing).
+    public static func createOrGet(with configuration: FlagsClient.Configuration, name: String, in core: DatadogCoreProtocol = CoreRegistry.default) -> FlagsClient {
```

Comment:
We decided on `createOrGet` in our meeting earlier today, but I wonder do we need `.create` and `.instance` and `createOrGet` to all be `public`?